packages-ts/gauntlet-terra-cw20-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,275 @@
+          msgs: [this.toWasmMsg(message)],
+          title: command.id,
+          // TODO: Set expiration time
+          // latest: { never: {} }
+        },
+      }
+      return new MsgExecuteContract(signer, this.multisig, proposeInput)
+    }
+
+    executeApproval: ProposalAction = async (signer, proposalId) => {
+      logger.info(`Generating data for approving proposal ${proposalId}`)
+      const approvalInput = {
+        vote: {
+          vote: Vote.YES,
+          proposal_id: proposalId,
+        },
+      }
+      return new MsgExecuteContract(signer, this.multisig, approvalInput)
+    }
+
+    executeExecution: ProposalAction = async (signer, proposalId) => {
+      logger.info(`Generating data for executing proposal ${proposalId}`)
+      const executeInput = {
+        execute: {
+          proposal_id: proposalId,
+        },
+      }
+      return new MsgExecuteContract(signer, this.multisig, executeInput)
+    }
+
+    fetchState = async (proposalId?: number): Promise<State> => {
+      const groupState = await this.query(this.multisigGroup, {
+        list_members: {},
+      })
+      const owners = groupState.members.map((m) => m.addr)
+      const thresholdState = await this.query(this.multisig, {
+        threshold: {},
+      })
+      const threshold = thresholdState.absolute_count.total_weight
+      if (!proposalId) {
+        return {
+          threshold,
+          nextAction: Action.CREATE,
+          owners,
+          approvers: [],
+        }
+      }
+      const proposalState = await this.query(this.multisig, {
+        proposal: {
+          proposal_id: proposalId,
+        },
+      })
+      const votes = await this.query(this.multisig, {
+        list_votes: {
+          proposal_id: proposalId,
+        },
+      })
+      const status = proposalState.status
+      const toNextAction = {
+        passed: Action.EXECUTE,
+        open: Action.APPROVE,
+        pending: Action.APPROVE,
+        rejected: Action.NONE,
+        executed: Action.NONE,
+      }
+      return {
+        threshold,
+        nextAction: toNextAction[status],
+        owners,
+        currentStatus: status,
+        data: proposalState.msgs,
+        approvers: votes.votes.filter((v) => v.vote === Vote.YES).map((v) => v.voter),
+      }
+    }
+
+    printPostInstructions = async (proposalId: number) => {
+      const state = await this.fetchState(proposalId)
+      const approvalsLeft = state.threshold - state.approvers.length
+      const messages = {
+        [Action.APPROVE]: `The proposal needs ${approvalsLeft} more approvals. Run the same command with the flag --proposal=${proposalId}`,
+        [Action.EXECUTE]: `The proposal reached the threshold and can be executed. Run the same command with the flag --proposal=${proposalId}`,
+        [Action.NONE]: `The proposal has been executed. No more actions needed`,
+      }
+      logger.line()
+      logger.info(`${messages[state.nextAction]}`)
+      logger.line()
+    }
```

Comment:
This is something we used to call an inspection command.

Here it's just a function, on a different command class. 
The Multisig command, that groups all Multisig abstract commands?

How do I call this inspection command? How do I compose it in a workflow?
Is the Multisig command a workflow or is it something else? What exactly?

I have many questions :)

...

The first priority is to have it function in any way possible, and while this looks ok IMO we need to go back to the Gauntlet v4 drawing board and start typing and composing these commands better.

Reply:
I see your point. This goes a bit beyond from this PR. Opening an issue for it: #133